app/models/manageiq/providers/embedded_terraform/automation_manager/job.rb

```diff
@@ -21,12 +23,28 @@ def pre_execute
   def execute
     template_path = File.join(options[:git_checkout_tempdir], template_relative_path)
     credentials   = Authentication.where(:id => options[:credentials])
-    extra_vars    = options.dig(:input_vars, :extra_vars) || {}
-
-    response = Terraform::Runner.run(decrypt_extra_vars(extra_vars), template_path, :credentials => credentials, :env_vars => options[:env_vars])
-
-    options[:terraform_stack_id] = response.stack_id
-    save!
+    input_vars    = options.dig(:input_vars, :extra_vars) || {}
+    action        = options[:action]
+
+    case action
+    when ResourceAction::RETIREMENT
```

Comment:
This looks like it could easily just be
```suggestion
    if action == ResourceAction::RETIREMENT
```

I don't see the case buying you anything